include/bluetooth/services/cts_client.h

```diff
@@ -0,0 +1,152 @@
+};
+
+/**@brief Data structure for the Current Time characteristic. */
+struct bt_cts_current_time_char {
+	struct bt_cts_exact_time_256 exact_time_256;
+	struct bt_cts_adjust_reason adjust_reason;
+};
+
+/**@brief Current Time Service client event type. */
+enum bt_cts_client_evt_type {
+	/** A new Current Time reading has been received. */
+	BT_CTS_CLIENT_EVT_CURRENT_TIME,
+
+	/** The Current Time value received from the peer is invalid. */
+	BT_CTS_CLIENT_EVT_INVALID_TIME
+};
+
+/**@brief Current Time Service client event. */
+struct bt_cts_client_evt {
+	/** Type of event. */
+	enum bt_cts_client_evt_type evt_type;
+
+	/** Current Time characteristic data. This is filled when the evt_type is @ref BT_CTS_CLIENT_EVT_CURRENT_TIME. */
+	struct bt_cts_current_time_char current_time;
+};
+
+struct bt_cts_client;
+
+/**@brief Current Time Service client event handler type. */
+typedef void (*bt_cts_client_evt_handler)(struct bt_cts_client *cts_c,
+					  const struct bt_cts_client_evt *evt);
+
+/**@brief Current Time Service client structure. This structure contains status information for the client. */
+struct bt_cts_client {
+	/** Connection object. */
+	struct bt_conn *conn;
+
+	/** Handle of the Current Time Characteristic. */
+	uint16_t handle_ct;
+
+	/** Handle of the CCCD of the Current Time Characteristic. */
+	uint16_t handle_ct_ccc;
+
+	/** GATT read parameters. */
+	struct bt_gatt_read_params read_params;
+
+	/** Semaphore for GATT read parameters. */
+	struct k_sem read_params_sem;
+
+	/** Event handler to be called for handling events from the Current Time Service client. */
+	bt_cts_client_evt_handler evt_handler;
+};
+
+/**@brief Function for initializing the Current Time Service client.
+ *
+ * @param[out] cts_c       Current Time Service client structure. This structure must
+ *                         be supplied by the application. It is initialized by this
+ *                         function and can later be used to identify this particular client
+ *                         instance.
+ * @param[in]  evt_handler Event handler for Current Time Service client.
+ *
+ * @retval 0 If the client was initialized successfully.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_cts_client_init(struct bt_cts_client *cts_c,
+		       bt_cts_client_evt_handler evt_handler);
+
+/**@brief Function for clearing the CTS client connection.
+ *
+ * @param[in] cts_c  Current Time Service client structure.
+ */
+void bt_cts_client_on_disconnected(struct bt_cts_client *cts_c);
+
+/**@brief Function for reading the peer's Current Time Service Current Time characteristic.
+ *
+ * @param[in] cts_c    Current Time Service client structure.
+ * @param[in] timeout  Time-out for taking GATT read parameters semaphore.
+ *
+ * @retval 0 If all operations are successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int bt_cts_client_current_time_read(struct bt_cts_client *cts_c,
```

Comment:
It's preferable to not use the `_client_` infix, and we add it only where we need it (for services), i.e where it would otherwise conflict with a server version of the same function, which is why it is cts_init for server, and cts_client_init for client.

Also read should be place in different order (similar as `bt_bas_read_battery_level`).

So this should be: `bt_cts_read_current_time`

Reply:
Will do.